src/Components/Form/FormFields/AutoExpandingTextInputFormField.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useRef } from "react";
+import TextAreaFormField, { TextAreaFormFieldProps } from "./TextAreaFormField";
+
+type AutoExpandingTextInputFormFieldProps = TextAreaFormFieldProps & {
+  maxHeight?: number;
+};
+
+const AutoExpandingTextInputFormField = (
+  props: AutoExpandingTextInputFormFieldProps
+) => {
+  const myref = useRef<HTMLTextAreaElement>(null);
+  useEffect(() => {
+    if (myref.current == null) return;
+    const text = myref.current.textContent?.split("\n");
+    const len = (text?.length == 0 ? 1 : text?.length) || 1;
+    const height = Math.min(len * 18, (props.maxHeight || 160) - 28) + 28;
```

Comment:
This is hard to maintain for someone else stumbling upon this. Add comment explaining what you are doing here. Why `18`; why subtract `28`; why add `28` at the end.

Reply:
@kshitijv256 instead of using haedcoded value when not get the line height from styles and then use it, just a thought